<!DOCTYPE html>
<html lang="en">
<head>
	<title>ModelTeller Overview</title>
	<!--<meta charset="utf-8">-->
	<!--<meta name="viewport" content="width=device-width, initial-scale=1">-->

	<link rel="shortcut icon" type="image/x-icon" href="https://modelteller.tau.ac.il/pics/logo.gif" />
	<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
	<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
	<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
	<link rel="stylesheet" href="https://gitcdn.github.io/bootstrap-toggle/2.2.2/css/bootstrap-toggle.min.css">

	<!--files for the select tag-->
	<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-select/1.12.4/css/bootstrap-select.min.css">
	<script src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-select/1.12.4/js/bootstrap-select.min.js"></script>
	<!--<script src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-select/1.12.4/js/i18n/defaults-*.min.js"></script>-->

	<!--files for the checkbox tag (toggle)-->
	<script src="https://gitcdn.github.io/bootstrap-toggle/2.2.2/js/bootstrap-toggle.min.js"></script>


	<link rel="stylesheet" href="css/general.css">
	<link rel="stylesheet" href="css/nav.css">

	<script src="https://code.jquery.com/jquery-1.10.2.js"></script>
</head>
<body>
<nav role="navigation" class="navbar navbar-inverse" id="nav">
    <div class="jumbotron" id="jumbo">
        <div class="container">
            <div class="row" id="title-row">
                <div class="col-md-1">
                </div>
                <div class="col-md-10" align="center">
                    <span id="server-title">ModelTeller</span><br>
                    <img src="pics/logo.gif" id="nav_bar_image" class="img-rounded"><br>
                    <span id="sub-title">A machine learning tool for phylogenetic model selection</span>
					<br>
					<span id="research-site">
					Research Site | <a href="https://m.tau.ac.il/~itaymay/index.html" target="_blank" style="color: inherit">Mayrose Group</a>
					</span>
					<br>
					<br>
                </div>
            </div>
            <div class="col-md-1">
            </div>
        </div>
    </div>
		<div class="container" id="nav_container">
			<div class="row">
				<div class="col-md-2">
				</div>
				<div class="col-md-8">
					<ul class="nav navbar-nav" id="menu-nav">
						<li>
							<a href="index.html">Home</a>
						</li>
						<li class="active">
							<a href="overview.html">Overview</a>
						</li>
						<li>
							<a href="download.html">Download</a>
						</li>
						<li>
							<a href="https://english.tau.ac.il/" id="tau-link" target="_blank">
								<img src=pics/TAU_LOGO.png id="tau-logo">
							</a>
						</li>
					</ul>
				</div>
				<div class="col-md-2">
				</div>
			</div>
		</div>
	</nav>
	<div id="behind-nav-bar">
	</div>
	<div class="container" id="overview_container">
		<div class="row">
			<div class="col-md-2">
			</div>
			<div class="col-md-8" align="justify">
				ModelTeller Q&As:
				<a href="#what_is_modelteller">What is ModelTeller?</a> |
				<a href="#input">What is the input?</a> |
				<a href="#Running_modes">Which running mode to use?</a> |
				<a href="#output">What is the output?</a> |
				<a href="#substitution_models">What are the candidate models?</a> |
				<a href="#Running time">What about running time?</a> |
				<a href="#ModelTeller accuracy">Is it accurate??</a>
				<br><br>

				<h2><div id="what_is_modelteller">What is ModelTeller?</div></h2>
				ModelTeller is a machine-learning algorithm optimized to predict <b>the nucleotide substitution model that yields the most accurate <u>branch-length estimates</u></b>. <br><br>
				ModelTeller is implemented using a Random Forest algorithm, trained on
				thousands of datasets that range over a wide range of empirical properties.
				In the implementation of ModelTeller, the propensity of every
				<a href="#substitution_models">substituion model</a> is computed and given as an
				<a href="#output">output</a>.
				All you need to do is upload the alignment or multiple consecutive alignments within a file or inside the textbox.
				ModelTeller will read your input, compute the relevant features, and report the best and alternative models for each of the alignments.
				Following the prediction of the best model, ModelTeller computes the maximum-likelihood tree for the selected model and your input data.
				<br><br><img src="pics/modelteller_flowchart.jpg" id="modelteller_flowchart" width="700">
				<br><br>
				<font size="3"><i><b>A schematic flowchart of the data and procedures applied for the study of ModelTeller.</b>
				The black solid arrows represent the computational pipeline and the dashed grey arrows represent
				a prediction scheme, either cross-validation or validation procedures.
				Given an empirical sequence database, we first computed the optimized parameters for each of
				<a href="#substitution_models">the 24 nucleotide substituion model</a>. Given these parameters, simulations were made to generate the training/validation data and informative features were extracted from the simulated alignments.
					The features of the training set were used to train
				ModelTeller and assess its performance in cross-validation. The features of the validation
				set were used for assessing the predictions of ModelTeller. The asterisk represents procedures
				that were done once for the analysis of <a href="#ModelTeller_option">ModelTeller</a> and once for <a href="#ModelTeller_G_option">ModelTeller<sub>G</sub></a>:
				for ModelTeller, the features were computed over a quick reconstruction of a BioNJ tree;
					for ModelTeller<sub>G</sub>, the GTR+I+G tree was first optimized, and the features were
				computed over this tree. The training and prediction were made according to the relevant sets
					of features. <a href="#ModelTeller accuracy">Click here for a quick view of the results!</a></i></font>
				<br><br>


				<h2><div id="input">Input sequence data</div></h2>
				Upload the alignment or multiple consecutive alignments within a file or inside the textbox.
				Acceptable input can be any alignment file format that is valid in Biopython (listed below):<br><br>
				<ul>
				<li><b>Clustal</b> - output from Clustal W or X.</li>
				<li><b>Emboss</b> - EMBOSS tools' "pairs" and "simple" alignment formats.</li>
				<li><b>Fasta</b> - the generic sequence file format where each record starts with an identifier line starting with a ">" character, followed by lines of sequence. <u>Note that multiple alignments are not acceptable in this format.</u></li>
				<li><b>Ig</b> - the IntelliGenetics file format, apparently the same as the MASE alignment format.</li>
				<li><b>Mauve</b> - output from progressiveMauve/Mauve.</li>
				<li><b>Nexus</b> - output from NEXUS.</li>
				<li><b>Phylip/phylip-sequential/phylip-relaxed.</b></li>
				<li><b>Stockholm</b> - a richly annotated alignment file format used by PFAM.</li><br>
				</ul>
				Once you upload the alignment, ModelTeller detects the format and continues the computation.<br>
				Only nucleotide sequences are valid. In case that ModelTeller could not read your input file, it would notify you with an appropriate error message.
				<br><br>
				
				<h2><div id="Running_modes">Running modes</div></h2>
				<ul>
				<li id="ModelTeller_option"><b>Select the best model for branch-lengths estimation (ModelTeller)</b> - this option will rapidly select a single model for maximum-likelihood phylogeny reconstruction. The maximum-likelihood phylogeny, i.e., the model parameters, the branch-lengths, and the topology will be optimized for you after the model is predicted.</li><br>
					<li id="ModelTeller_G_option"><b>Use a fixed GTR+I+G topology (ModelTeller<sub>G</sub>)</b> - this option will first optimize the GTR+I+G phylogeny and then predict the optimal model for branch-length estimation. The maximum-likelihood phylogeny will be optimized given the GTR+I+G fixed topology. Prediction of the best model might take a little while due to the preliminary GTR+I+G phylogeny reconstruction, but the final tree will be computed rapidly. This procedure was demonstrated to perform best both for topologies and branch-length estimation accuracies on complex simulation models.</li><br>
				<li><b>User defined topology</b> - if you have a good, validated topology for your data, please provide it and ModelTeller will predict the best model for branch-length estimation. The maximum-likelihood phylogeny will be computed for you given your fixed topology. In case you provided multiple alignments, multiple trees should be provided in the input file, each in a separate line. The input tree should be in Newick format and the IDs of the tips must match the IDs in the sequence alignment.</li>
				</ul>
				<br><br><br><br><br><br><br><br>

				<h2><div id="output">Output</div></h2>
                <ul>
                    1. The <b>best predicted model</b> and the ranking of alternative models, in case that you cannot use the best model in a downstream application.<br><br>
                    2. The <b>maximum-likelihood tree</b>, reconstructed using the best model.<br><br>
					3. Interested in the features that led to this prediction? If you check the <b>"Compute features contribution"</b> checkbox, a csv file will be available for download after the tree is computed. Each entry in the file represents the weight of a single feature in computing the probability of choosing one model, as extracted from the paths along the Random Forest implementation.<br>
                </ul>
				<br>

				<h2><div id="substitution_models">The substitution models</div></h2>
				ModelTeller will return the best model, and the ranking of alternative models among the following 24 nucleotide substitution models: <br>
				<ul>
				<li> Jukes and Cantor 1969 (<b>JC</b>)</li>
				<li> Felsenstein 1981 (<b>F81</b>)</li>
				<li> Kimura two parameters 1980 (<b>K2P</b>)</li>
				<li> Hasegawa-Kishino-Yano 1985 (<b>HKY</b>)</li>
				<li> The generalized symmetrical model, Zharkikh, 1994 (<b>SYM</b>)</li>
				<li> General Time Reversible, Tavare 1986 (<b>GTR</b>)</li><br>
				</ul>
				combined with the proportion of invariable sites (<b>+I</b>), rate heterogeneity across sites (<b>+G</b>), or both (<b>+I+G</b>).
				<br><br>


				<h2><div id="Running time">Running time</div></h2>
				Whereas the statistical methods for phylogenetic reconstruction may take hours or even days of computation, ModelTeller retrieves results instantly!
				Computation of the maximum-likelihood tree is done by PhyML after the best model is predicted and might take a little longer.
				<br><br>


				<h2><div id="ModelTeller accuracy">ModelTeller accuracy</div></h2>
				ModelTeller is optimized to retrieve the model that generates the most accurate branch-length estimates. ModelTeller was examined both in cross-validation on the training set <i>T<sub>1</sub></i> and on the validation set, <i>V</i>.
				The figures below demonstrate ModelTeller's performance compared to AIC, BIC, using the most complex model, GTR+I+G, consistently, and selecting a model at random for each dataset.
				<img src="pics/modelteller_violinplots.jpg" id="accuracy_img" width="750">
				<br>
				<font size="3">
				<i><b>Distribution of the true ranks of the selected models.</b>
					The x axis represents five methodologies for model selection:
					ModelTeller, AIC, BIC, consistently using the GTR+I+G model, and selecting a model at random
					for each datasets. The y axis represents the true ranks (from 1 to 24)
					of the models selected by the methodologies according to the observed branch-length error
					(where 1 and 24 represnt the most and least accurate models, respectively).
					Left panel: distribution over the T<sub>1</sub> set with medians: 8 for ModelTeller and BIC,
					9 for AIC and GTR+I+G, and 13 for a random selection of model.
					Right panel: distribution over V with medians: 10 for ModelTeller, 13 for AIC, BIC,
					and for a random selection of model, and 15 for GTR+I+G. The black horizontal lines represent
					the medians, the thick white bars represent the interquartile (IQR) range, and the thin white
					lines extend beyond 1.5*the IQR range. The violin plots represent a kernel density estimation of the underlying distribution of the true rank of the selected models; wider sections of the violin plot represent a higher probability that the methodology selects models of that rank while skinnier sections represent a lower probability.
				</i></font><br><br>
				<br><br>
				<br><br>


			</div>
            <div class="col-md-2">
            </div>
		</div>
	</div>
        <nav role="navigation" class="navbar-inverse navbar-fixed-bottom ng-binding">
            Research Site | <a href="https://m.tau.ac.il/~itaymay/index.html" target="_blank">Mayrose Group</a>
        </nav>
</body>
</html>







