<!DOCTYPE html>
<html lang="en">
<head>
	<title>ModelTeller Overview</title>
	<!--<meta charset="utf-8">-->
	<!--<meta name="viewport" content="width=device-width, initial-scale=1">-->

	<link rel="shortcut icon" type="image/x-icon" href="https://modelteller.tau.ac.il/pics/logo.gif" />
	<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
	<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
	<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
	<link rel="stylesheet" href="https://gitcdn.github.io/bootstrap-toggle/2.2.2/css/bootstrap-toggle.min.css">

	<!--files for the select tag-->
	<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-select/1.12.4/css/bootstrap-select.min.css">
	<script src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-select/1.12.4/js/bootstrap-select.min.js"></script>
	<!--<script src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-select/1.12.4/js/i18n/defaults-*.min.js"></script>-->

	<!--files for the checkbox tag (toggle)-->
	<script src="https://gitcdn.github.io/bootstrap-toggle/2.2.2/js/bootstrap-toggle.min.js"></script>


	<link rel="stylesheet" href="css/general.css">
	<link rel="stylesheet" href="css/nav.css">

	<script src="https://code.jquery.com/jquery-1.10.2.js"></script>
</head>
<body>
<nav role="navigation" class="navbar navbar-inverse" id="nav">
    <div class="jumbotron" id="jumbo">
        <div class="container">
            <div class="row" id="title-row">
                <div class="col-md-1">
                </div>
                <div class="col-md-10" align="center">
                    <span id="server-title">ModelTeller</span><br>
                    <img src="pics/logo.gif" id="nav_bar_image" class="img-rounded"><br>
                    <span id="sub-title">A machine learning tool for phylogenetic model selection</span>
					<br><br>
                </div>
            </div>
            <div class="col-md-1">
            </div>
        </div>
    </div>
		<div class="container" id="nav_container">
			<div class="row">
				<div class="col-md-2">
				</div>
				<div class="col-md-8">
					<ul class="nav navbar-nav" id="menu-nav">
						<li>
							<a href="index.html">Home</a>
						</li>
						<li class="active">
							<a href="overview.html">Overview</a>
						</li>
						<li>
							<a href="download.html">Download</a>
						</li>
						<li>
							<a href="https://english.tau.ac.il/" id="tau-link" target="_blank">
								<img src=pics/TAU_LOGO.png id="tau-logo">
							</a>
						</li>
					</ul>
				</div>
				<div class="col-md-2">
				</div>
			</div>
		</div>
	</nav>
	<div id="behind-nav-bar">
	</div>
	<div class="container" id="overview_container">
		<div class="row">
			<div class="col-md-2">
			</div>
			<div class="col-md-8" align="justify">
				<h2>Phylogenetic  Model  Selection</h2>
				Phylogenetic Model Selection is considered as a mandatory step for phylogenetic tree reconstruction. <br>
				The current habit is to compute the likelihood of every candidate model to the data, and eventually select the best-fitted one. However, <br>
				<ul>
				<li>Current model selection criteria (e.g., AIC, BIC, hLRT) were not optimized for phylogenetic data</li>
				<li>They detect the best model for inference but not the best model for accurate phylogenetic reconstruction</li>
				<li>The computation is extremely long!</li><br>
				</ul>

				<h2>What is ModelTeller?</h2>
				ModelTeller is a machine learning algorithm that is optimized to predict <b>the nucleotide substitution model that yields the most accurate tree</b>. <br><br>
				ModelTeller was trained across 11,000 alignments that were sampled from empirical cases. These range across a wide range of sequence lengths, number of sequences, and sequence divergences. ModelTeller was optimized to predict the model that generates a tree of lowest distance to the true one (in simulations).
				<br><br>


				<h2>How to use ModelTeller?</h2>
				All you need to do is upload the alignment or multiple consecutive alignments within a file or inside the textbox. 
				ModelTeller will read your input, compute the relevant features, and report the best and alternative models for each of the alignments.
				<br><br>


				<h2>Input<br></h2>
				ModelTeller is implemented in python and accepts as input any alignment file format that is valid in Biopython (listed below). Once you upload the alignment, ModelTeller detects the format and continues the computation:<br><br>
				<ul>
				<li><b>Clustal</b> - output from Clustal W or X.</li>
				<li><b>Emboss</b> - EMBOSS tools' "pairs" and "simple" alignment formats.</li>
				<li><b>Fasta</b> - the generic sequence file format where each record starts with an identifier line starting with a ">" character, followed by lines of sequence. <b>Note that multiple alignments are not acceptable in this format.</b></li>
				<li><b>Ig</b> - the IntelliGenetics file format, apparently the same as the MASE alignment format.</li>
				<li><b>Mauve</b> - output from progressiveMauve/Mauve.</li>
				<li><b>Nexus</b> - output from NEXUS.</li>
				<li><b>Phylip/phylip-sequential/phylip-relaxed.</b></li>
				<li><b>Stockholm</b> - a richly annotated alignment file format used by PFAM.</li><br>
				</ul>
				In case that ModelTeller could not read your input file, it would notify you with an error message.
				<br><br>


				<h2>The substitution models</h2>
				ModelTeller incorporates 24 nucleotide substitution models. Those include: <br>
				<ul>
				<li> Jukes and Cantor 1969 (<b>JC</b>)</li>
				<li> Felsenstein 1981 (<b>F81</b>)</li>
				<li> Kimura two parameters 1980 (<b>K2P</b>)</li>
				<li> Hasegawa-Kishino-Yano 1985 (<b>HKY</b>)</li>
				<li> The generalized symmetrical model, Zharkikh, 1994 (<b>SYM</b>)</li>
				<li> General Time Reversible, Tavare 1986 (<b>GTR</b>)</li><br>
				</ul>
				combined with the proportion of invariable sites (<b>+I</b>), rate heterogeneity across sites (<b>+G</b>), or both (<b>+I+G</b>).
				<br><br>


				<h2>ModelTeller running time</h2>
				Whereas the statistical methods for phylogenetic reconstruction may take hours or even days of computation, ModelTeller retrieves results instantly! 
				<br><br>


				<h2>ModelTeller accuracy</h2>
				ModelTeller is optimized to retrieve the most accurate phylogenetic tree.
				The figure below demonstrates ModelTeller's performance compared to AIC, BIC, and using always the most complex model, GTR+I+G.
				<img src="pics/modtel_accuracy.jpg" id="overview_picture">
				<br>
				The y axis represents the true ranks of the selected models by each of the model selection strategies. 
				The white dots represent the median, the thick gray bars represent the interquartile (IQR) range, and the thin gray lines extend beyond 1.5*the IQR range. 
				The violin plots represent a kernel density estimation of the underlying distribution of the true rank of the selected models; 
				wider sections of the violin plot represent a higher probability that the strategy selects models of that rank while skinnier sections represent a lower probability. 
				
				<br><br>


			

				<!--<li><a href="http://www.pnas.org/content/111/6/2259" target="_blank" id="ref1">Lavinder, J. J. et al. <i>Proc Natl Acad Sci USA</i> 111, 2259-2264 (2014).</a></li>-->
				<!--<br>-->
				<!--<li><a href="http://www.pnas.org/content/110/8/2993" target="_blank" id="ref2">Wine, Y. et al. <i>Proc Natl Acad Sci USA</i> 110, 2993-2998 (2013).</a></li>-->
				<!--<br>-->
				<!--<li><a href="http://www.nature.com/articles/nbt.2782" target="_blank" id="ref3">Georgiou, G. et al. <i>Nat Biotech</i> 32, 158-168 (2014).</a></li>-->
				<!--<br>-->
				<!--<li><a href="https://www.sciencedirect.com/science/article/pii/S1074761300000066" target="_blank" id="ref4">Xu, J. L. et al. <i>Immunity</i> 13, 37-45 (2000).</a></li>-->
				<!--<br>-->
				<!--<li><a href="https://www.ncbi.nlm.nih.gov/books/NBK10757/" target="_blank" id="ref5">Murphy, K. et al. <i>Janeway's immunobiology</i>.  (Garland Science, 2012).</a></li>-->
				<!--<br>-->
				<!--<li><a href="https://academic.oup.com/nar/article/27/1/209/1234872" target="_blank" id="ref6">Lefranc, M.-P. et al. <i>Nucleic Acids Research</i> 27, 209-212 (1999).</a></li>-->
				<!--<br>-->
				<!--<li><a href="https://www.nature.com/articles/nmeth.3364" target="_blank" id="ref7">Bolotin, D. A. et al. <i>Nature Methods</i> 12, 380 (2015).</a></li>-->
				<br><br>


			</div>
            <div class="col-md-2">
            </div>
		</div>
	</div>
        <nav role="navigation" class="navbar-inverse navbar-fixed-bottom ng-binding">
            Research Site | <a href="https://m.tau.ac.il/~itaymay/index.html" target="_blank">Mayrose Group</a>
        </nav>
</body>
</html>







