<!DOCTYPE html>
<html lang="en">
<head>
	<title>ModelTeller Overview</title>
	<!--<meta charset="utf-8">-->
	<!--<meta name="viewport" content="width=device-width, initial-scale=1">-->

	<link rel="shortcut icon" type="image/x-icon" href="https://modelteller.tau.ac.il/pics/logo.gif" />
	<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
	<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
	<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
	<link rel="stylesheet" href="https://gitcdn.github.io/bootstrap-toggle/2.2.2/css/bootstrap-toggle.min.css">

	<!--files for the select tag-->
	<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-select/1.12.4/css/bootstrap-select.min.css">
	<script src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-select/1.12.4/js/bootstrap-select.min.js"></script>
	<!--<script src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-select/1.12.4/js/i18n/defaults-*.min.js"></script>-->

	<!--files for the checkbox tag (toggle)-->
	<script src="https://gitcdn.github.io/bootstrap-toggle/2.2.2/js/bootstrap-toggle.min.js"></script>


	<link rel="stylesheet" href="css/general.css">
	<link rel="stylesheet" href="css/nav.css">

	<script src="https://code.jquery.com/jquery-1.10.2.js"></script>
</head>
<body>
<nav role="navigation" class="navbar navbar-inverse" id="nav">
    <div class="jumbotron" id="jumbo">
        <div class="container">
            <div class="row" id="title-row">
                <div class="col-md-1">
                </div>
                <div class="col-md-10" align="center">
                    <span id="server-title">ModelTeller</span><br>
                    <img src="pics/logo.gif" id="nav_bar_image" class="img-rounded"><br>
                    <span id="sub-title">A machine learning tool for phylogenetic model selection</span>
					<br>
					<span id="research-site">
					Research Site | <a href="https://m.tau.ac.il/~itaymay/index.html" target="_blank" style="color: inherit">Mayrose Group</a>
					</span>
					<br>
					<br>
                </div>
            </div>
            <div class="col-md-1">
            </div>
        </div>
    </div>
		<div class="container" id="nav_container">
			<div class="row">
				<div class="col-md-2">
				</div>
				<div class="col-md-8">
					<ul class="nav navbar-nav" id="menu-nav">
						<li>
							<a href="index.html">Home</a>
						</li>
						<li class="active">
							<a href="overview.html">Overview</a>
						</li>
						<li>
							<a href="download.html">Download</a>
						</li>
						<li>
							<a href="https://english.tau.ac.il/" id="tau-link" target="_blank">
								<img src=pics/TAU_LOGO.png id="tau-logo">
							</a>
						</li>
					</ul>
				</div>
				<div class="col-md-2">
				</div>
			</div>
		</div>
	</nav>
	<div id="behind-nav-bar">
	</div>
	<div class="container" id="overview_container">
		<div class="row">
			<div class="col-md-2">
			</div>
			<div class="col-md-8" align="justify">
				<h2>What is ModelTeller?</h2>
				ModelTeller is a machine learning algorithm optimized to predict <b>the nucleotide substitution model that yields the most accurate branch-lengths estimates</b>. <br><br>
				ModelTeller was trained on thousands of datasets that range over a wide range of empirical properties, therefore it is suitable for many relevant datasets.
				<br>
				All you need to do is upload the alignment or multiple consecutive alignments within a file or inside the textbox. 
				ModelTeller will read your input, compute the relevant features, and report the best and alternative models for each of the alignments.
				Following the prediction of the best model, ModelTeller computes the maximum-likelihood tree for the selected model and your input data.
				<br><br>


				<h2>Input<br></h2>
				ModelTeller is implemented in python and accepts as input any alignment file format that is valid in Biopython (listed below). Once you upload the alignment, ModelTeller detects the format and continues the computation:<br><br>
				<ul>
				<li><b>Clustal</b> - output from Clustal W or X.</li>
				<li><b>Emboss</b> - EMBOSS tools' "pairs" and "simple" alignment formats.</li>
				<li><b>Fasta</b> - the generic sequence file format where each record starts with an identifier line starting with a ">" character, followed by lines of sequence. <b>Note that multiple alignments are not acceptable in this format.</b></li>
				<li><b>Ig</b> - the IntelliGenetics file format, apparently the same as the MASE alignment format.</li>
				<li><b>Mauve</b> - output from progressiveMauve/Mauve.</li>
				<li><b>Nexus</b> - output from NEXUS.</li>
				<li><b>Phylip/phylip-sequential/phylip-relaxed.</b></li>
				<li><b>Stockholm</b> - a richly annotated alignment file format used by PFAM.</li><br>
				</ul>
				Only nucleotide sequences are valid. In case that ModelTeller could not read your input file, it would notify you with an error message.
				<br><br>
				
				<h2>Running modes<br></h2>
				<ul>
				<li><b>Select the best model for branch-lengths estimation</b> - this option will rapidly select a single model for maximum-likelihood phylogeny reconstruction. The maximum-likelihood phylogeny, i.e., the model parameters, the branch-lengths, and the topology will be optimized for you after the model is predicted.</li>
				<li><b>Use a fixed GTR+I+G topology</b> - this option will first optimized the GTR+I+G phylogeny and then predict the optimal model for branch-length estimation. The maximum-likelihood phylogeny will be optimized given the GTR+I+G fixed topology. Prediction of the best model might take a little while due to the preliminary GTR+I+G phylogeny reconstruction, but the final tree will be computed rapidly. This procedure was demonstrated to perform best both for topologies and branch-length estimation accuracies on complex simulation models.</li>
				<li><b>User defined topology</b> - if you have a good, validated topology for your data, please provide it and ModelTeller will predict the best model for branch-length estimation. The maximum-likelihood phylogeny will be computed for you given your fixed topology. In case you provided multiple alignments, multiple trees should be provided in the input file, each in a separate line. The input tree should be in Newick format and the IDs of the tips must match the IDs in the sequence alignment.</li>
				</ul>
				<br><br><br><br><br><br><br><br>


				<h2>The substitution models</h2>
				ModelTeller will return the best model, and the ranking of alternative models among the following 24 nucleotide substitution models: <br>
				<ul>
				<li> Jukes and Cantor 1969 (<b>JC</b>)</li>
				<li> Felsenstein 1981 (<b>F81</b>)</li>
				<li> Kimura two parameters 1980 (<b>K2P</b>)</li>
				<li> Hasegawa-Kishino-Yano 1985 (<b>HKY</b>)</li>
				<li> The generalized symmetrical model, Zharkikh, 1994 (<b>SYM</b>)</li>
				<li> General Time Reversible, Tavare 1986 (<b>GTR</b>)</li><br>
				</ul>
				combined with the proportion of invariable sites (<b>+I</b>), rate heterogeneity across sites (<b>+G</b>), or both (<b>+I+G</b>).
				<br><br>


				<h2>Running time</h2>
				Whereas the statistical methods for phylogenetic reconstruction may take hours or even days of computation, ModelTeller retrieves results instantly! <br>
				Computation of the maximum-likelihood tree is done by PhyML after the best model is predicted and might take a little longer.
				<br><br>


				<h2>ModelTeller accuracy</h2>
				ModelTeller is optimized to retrieve the model that generates the most accurate branch-length estimates.
				The figure below demonstrates ModelTeller's performance compared to AIC, BIC, and using the most complex model, GTR+I+G, consistently.
				<img src="pics/modtel_accuracy.jpg" id="overview_picture">
				<br>
				The y axis represents the true ranks of the selected models by each of the model selection strategies. 
				The white dots represent the median, the thick grey bars represent the interquartile (IQR) range, and the thin gray lines extend beyond 1.5*the IQR range. 
				The violin plots represent a kernel density estimation of the underlying distribution of the true rank of the selected models; 
				wider sections of the violin plot represent a higher probability that the strategy selects models of that rank while skinnier sections represent a lower probability. 
				
				<br><br>
				<br><br>


			</div>
            <div class="col-md-2">
            </div>
		</div>
	</div>
        <nav role="navigation" class="navbar-inverse navbar-fixed-bottom ng-binding">
            Research Site | <a href="https://m.tau.ac.il/~itaymay/index.html" target="_blank">Mayrose Group</a>
        </nav>
</body>
</html>







